hack/lib/dev-env.sh

```diff
@@ -0,0 +1,182 @@
+kind_config_host_dir="${OUTPUT}/dev/kind"
+kind_config="${kind_config_host_dir}/env.yaml"
+authenticator_config_host_dir="${OUTPUT}/dev/authenticator/config"
+authenticator_config="${authenticator_config_host_dir}/authenticator.yaml"
+authenticator_export_host_dir="${OUTPUT}/dev/authenticator/export"
+authenticator_state_host_dir="${OUTPUT}/dev/authenticator/state"
+authenticator_config_dest_dir="/etc/authenticator"
+authenticator_export_dest_dir="/var/authenticator/export"
+authenticator_state_dest_dir="/var/authenticator/state"
+apiserver_config_dest_dir="/etc/kubernetes/authenticator"
+# Kubeconfig used when authenticator loads its mapping configuration from the API server
+authenticator_kubeconfig="${authenticator_config_dest_dir}/authenticator-kubeconfig.yaml"
+# Kubeconfig passed to the apiserver so it can kind its authentication webhook
+authenticator_generate_kubeconfig="${authenticator_export_dest_dir}/apiserver-auth-webhook-kubeconfig.yaml"
+apiserver_auth_webhook_kubeconfig="${apiserver_config_dest_dir}/apiserver-auth-webhook-kubeconfig.yaml"
+host_path_webhook_kubeconfig="${authenticator_export_host_dir}/apiserver-auth-webhook-kubeconfig.yaml"
+# Kubectl kubeconfig template
+kubectl_kubeconfig_template="${REPO_ROOT}/hack/dev/kubeconfig.yaml"
+client_dir="${OUTPUT}/dev/client"
+# Generated kubeconfig for testing
+kubectl_kubeconfig="${client_dir}/kubeconfig.yaml"
+# Admin kubeconfig generated by kind
+kind_kubeconfig="${client_dir}/kind-kubeconfig.yaml"
+
+function create_network() {
+    if [[ -n $(docker network ls | grep authenticator-dev) ]]; then
+        echo "${NETWORK_NAME} exists"
+    else
+        echo "Creating network ${NETWORK_NAME}"
+        docker network create "${NETWORK_NAME}" \
+            --driver bridge \
+            --subnet "${NETWORK_SUBNET}" \
+            --opt "com.docker.network.bridge.enable_ip_masquerade=true"
+    fi
+}
+
+function delete_network() {
+    if [[ -n $(docker network ls | grep authenticator-dev) ]]; then
+        echo "Deleting network ${NETWORK_NAME}."
+        docker network rm "${NETWORK_NAME}"
+    else
+        echo "Network ${NETWORK_NAME} doesn't exist, doing nothing."
+    fi
+}
+
+function write_authenticator_config() {
+    mkdir -p "${authenticator_config_host_dir}"
+    sed -e "s|{{ADMIN_ARN}}|${ADMIN_ARN}|g" \
+        -e "s|{{AUTHENTICATOR_STATE_DIR}}|${authenticator_state_dest_dir}|g" \
+        -e "s|{{AUTHENTICATOR_GENERATE_KUBECONFIG}}|${authenticator_generate_kubeconfig}|g" \
+        -e "s|{{AUTHENTICATOR_KUBECONFIG}}|${authenticator_kubeconfig}|g" \
+        -e "s|{{AUTHENTICATOR_PORT}}|${AUTHENTICATOR_PORT}|g" \
+        -e "s|{{AUTHENTICATOR_IP}}|${AUTHENTICATOR_IP}|g" \
+        -e "s|{{CLUSTER_NAME}}|${CLUSTER_NAME}|g" \
+        "${authenticator_config_template}" > "${authenticator_config}"
+}
+
+function write_kind_config() {
+    mkdir -p "${kind_config_host_dir}"
+    sed -e "s|{{AUTHENTICATOR_EXPORT_HOST_DIR}}|${authenticator_export_host_dir}|g" \
```

Comment:
do these commands work on mac? i know sed has different syntax for mac and linux. I suggest to keep them close, we can mandate installation of `gsed`?

Reply:
Yeah, I have no idea if they work on Mac but I've definitely ran into problems with sed commands that were written by mac users :)